src/object/srv_obj.c

```diff
@@ -93,10 +93,10 @@ obj_rw_complete(crt_rpc_t *rpc, struct obj_io_context *ioc,
 		if (update) {
 			rc = dtx_sub_init(dth, &orwi->orw_oid,
 					  orwi->orw_dkey_hash);
-			if (rc == 0)
-				rc = vos_update_end(ioh, ioc->ioc_map_ver,
-						    &orwi->orw_dkey, status,
-						    &ioc->ioc_io_size, dth);
+			rc = vos_update_end(ioh, ioc->ioc_map_ver,
```

Comment:
I wonder if this shouldn't be something like

if (status == 0)
    status = dtx_sub_init(dth, ....);
 rc = vos_update_end(...);

Reply:
Right, if "status" is non-zero, then no need to call dtx_sub_init().

Reply:
Thanks!